srcpkgs/gnutls/patches/modules-gnutls-fwd-decls.patch

```diff
@@ -0,0 +1,48 @@
+--- guile/modules/gnutls.in.orig
++++ guile/modules/gnutls.in
+@@ -566,20 +566,33 @@
+ 
```

Comment:
Where is this patch from? What does it do? Is it going to be upstreamed?

Reply:
> Where is this patch from? What does it do?

This patch (written by me) is to prevent errors when cross compiling. I make it like [that](https://gitlab.com/gnutls/gnutls/-/commit/676ae412aeea58fa2c6aa7960ec16223101556eb#c320a9515fbbbdc26cb6510f8ab9a0b169ac88b9):
```diff
@@ -521,8 +521,9 @@
     (or (getenv "GNUTLS_GUILE_EXTENSION_DIR")
         "@guileextensiondir@"))
 
-  (load-extension (string-append %libdir "/guile-gnutls-v-2")
-                  "scm_init_gnutls"))
+  (unless (getenv "GNUTLS_GUILE_CROSS_COMPILING")
+    (load-extension (string-append %libdir "/guile-gnutls-v-2")
+                    "scm_init_gnutls")))
```

> Is it going to be upstreamed?

Hmm, I didn't think about it, really. I'll try to send the patch to the upstream developers.

Reply:
Unfortunately, as long as I know there is no progress yet.
For reference, here's the link to the issue on gitlab: https://gitlab.com/gnutls/gnutls/-/issues/1137